src/extensions/locale/ext-locale-init.reb

```diff
@@ -453,34 +454,40 @@ unless 'Windows = first system/platform [
 ]
 
     hijack 'locale function [
+        {parameter of 'language/language* or 'territory/territory* returns a value from the ISO tables}
         type [word!]
         <has>
         iso-639 (iso-639-table)
         iso-3166 (iso-3166-table)
+        error-msg
     ][
         env-lang: get-env "LANG"
+        error-msg: spaced ["*** Unrecognised env LANG:" env-lang " - defaulting to US English ***"]
         unless env-lang [
             return _
         ]
 
         letter: charset [#"a" - #"z" #"A" - #"Z"]
         unless parse env-lang [
             copy lang: [some letter]
-            [#"_" copy territory: [some letter] | #"." (territory: copy "US")]
-            to end (if lang = "C" [lang: copy "en"])
+            #"_" copy territory: [some letter]
+            to end
         ][
-            fail spaced ["Non-standard env LANG:" env-lang]
+            if lang <> "C" [print error-msg]
```

Comment:
I believe "POSIX" is also a valid setting that's not in the ISO tables: http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html